src/routes/Signup.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+
+
+export default function Signup() {
+  const [username, setUsername] = useState('');
+  const [dateOfBirth, setDOB] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confPassword, setConfPassword] = useState('');
+  const [interests, setInterests] = useState('');
+
+  // function to update state of username with
+  // value enter by user in form
+  const handleUsernameChange = (e) => {
+    setUsername(e.target.value);
+  }
+  // function to update state of date of birth (DOB) with value
+  // enter by user in form
+  const handleDOBChange = (e) => {
+    setDOB(e.target.value);
+  }
+  // function to update state of email with value
+  // enter by user in form
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  }
+  // function to update state of password with
+  // value enter by user in form
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+  }
+  // function to update state of confirm password
+  // with value enter by user in form
+  const handleConfPasswordChange = (e) => {
+    setConfPassword(e.target.value);
+  }
+
+  // function to update state of interests
+  // with value enter by user in form
+  const handleInterestsChange = (e) => {
+    setInterests(e.target.value);
+  }
+  // below function will be called when user
+  // click on submit button .
+  const handleSubmit = (e) => {
+    if (password != confPassword) {
```

Comment:
I did some research, and this shouldn't matter, but I made the change anyway.

Essentially, !== and === compares the actual objects as well as their values. So for example

0=='0' is true but
0==='0' is false

because the double equals will cast the objects to see if they are the same.